### DISCUSSION TITLE
How to update AsyncValue partially

### DISCUSSION THREAD

**Discussion by yknevenky** [rrousselGit/riverpod#3586] [Q&A]
I am trying to update the state of a FamilyAsyncNotifier partially, I just want to know whether I am doing it right. Also in this case of updating a asyncvalue partially, I want to know what's differences in using whenData, AsyncGuard.value and update method . It would be a immense help, if you provide some clarity on this.

I have seen other discussions talking about using whenData, AsyncGuard.value and update method , but it's still not clear enough. If someone can help me on this, it would be very helpful. Although It would be great to listen from the author itself @rrousselGit .

Example

The state here is a freezed class and looks something like this 

College(
name: "",
students: [],
faculties: [],
departments: []
)

All the above properties are not updated in a one go, for example the below code tries to update the students property while retaining the other values to its previous value.

```
update((state) async {
      List<Student> students =
          await getStudentsByCollegeId(id: this.arg);
      return state.copyWith(students: students);
});
```

Am I doing it right or it needs correction.

**Comment by normidar** [comment]
You should know `copyWith` method will create a new data class, so you are not update partially.
